ComponentKit/TransactionalDataSources/Mac/CKNSTableViewDataSource.h

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <Cocoa/Cocoa.h>
+
+#import <ComponentKit/CKPlatform.h>
+
+
+#import <ComponentKit/CKMacros.h>
+#import <ComponentKit/CKDimension.h>
+
+#import <ComponentKit/CKTransactionalComponentDataSource.h>
+
+@protocol CKComponentProvider;
+@class CKTransactionalComponentDataSourceChangeset;
+
+/**
+ This class is an implementation of a `NSTableViewDataSource` and `NSTableViewDataDelegate` that can be used along with components. For each set of changes (i.e insertion/deletion/update
+ of items and/or insertion/deletion of sections) the datasource will compute asynchronously on a background thread the corresponding component trees and then
+ apply the corresponding UI changes to the table view leveraging automatically view reuse.
+ 
+ Doing so this reverses the traditional approach for a `NSTableViewDataSource`. Usually the controller layer will *tell* the `UICollectionView` to update and
+ then the `NSTableView` *ask* the datasource for the data. Here the model is more Reactive, from an external prospective, the datasource is *told* what
+ changes to apply and then *tell* the collection view to apply the corresponding changes.
+ */
+@interface CKNSTableViewDataSource : NSObject <NSTableViewDataSource, NSTableViewDelegate>
+
+/**
+ Designated initializer.
+
+ 
+ @param componentProvider Class implementing the pure function turning a model into components.@see CKComponentProvider.
+ @param context Will be passed to your componentProvider. @see CKComponentProvider.
+ */
+- (instancetype)initWithTableView:(NSTableView *)tableView
+                componentProvider:(Class<CKComponentProvider>)componentProvider
+                          context:(id<NSObject>)context;
+
+
+/**
+ * TODO: document
+ */
+- (void)applyChangeset:(CKTransactionalComponentDataSourceChangeset *)changeset
+                  mode:(CKTransactionalComponentDataSourceMode)mode
+              userInfo:(NSDictionary *)userInfo;
```

Comment:
That would be cool. You usually make the various columns from the same model object for each row, so shouldn't be too hard to support multiple columns. Nonetheless, most recent mac apps don't actually use the columns most of the time.